class-wc-calypso-bridge-shared.php

```diff
@@ -123,6 +125,16 @@ public function add_scripts() {
 			'version'                      => WC_CALYPSO_BRIDGE_CURRENT_VERSION,
 		);
 
+		if ( $is_ecom_trial ) {
+			$woo_express_introductory_offers = WC_Calypso_Bridge_Woo_Express_Introductory_offers::get_offers_for_current_blog();
+			if ( count( $woo_express_introductory_offers ) ) {
+				$woo_express_introductory_offers = array_filter( $woo_express_introductory_offers, function( $offer ) {
+					return in_array( $offer['product_slug'], WC_Calypso_Bridge_Woo_Express_Introductory_offers::WOO_EXPRESS_PRODUCT_SLUGS );
+				} );
+				$params['wooExpressIntroductoryOffer'] = WC_Calypso_Bridge_Woo_Express_Introductory_offers::extract_offer_data_for_js( current( $woo_express_introductory_offers ) );
```

Comment:
It seems that in certain cases `$woo_express_introductory_offers` may be an empty array due to the filter. Since `current` will return `false` when array is empty, should we add a check for this to ensure it return an expected array? Or should we move this logic into `get_offers_for_current_blog`?

Reply:
Good catch @chihsuan 👍 

I've updated `get_offers_for_current_blog` to accept a filter to reduce code in `class-wc-calypso-bridge-shared.php` and avoid using count() again. The current() should always return a value now. 

https://github.com/Automattic/wc-calypso-bridge/pull/1397/commits/017b821417b541c35d7fe672cd2d17351b5f7f90